File: backend-bot/src/main/java/cc/jktu/apps/bot/config/BotLifeCycle.java
package cc.jktu.apps.bot.config;

import cc.jktu.apps.bot.service.BotService;
import org.springframework.context.SmartLifecycle;
import org.springframework.stereotype.Component;

@Component
public class BotLifeCycle implements SmartLifecycle {

    private final BotService botService;

    public BotLifeCycle(BotService botService) {
        this.botService = botService;
    }

    @Override
    public void start() {
        botService.getBot().login();
        botService.registerHandlers();
    }

    @Override
    public void stop() {
        botService.getBot().close();
    }

    @Override
    public boolean isRunning() {
        return botService.getBot().isOnline();
    }

}

File: backend-bot/src/main/java/config/BotLifeCycle.kt
package cc.jktu.apps.bot.config

import cc.jktu.apps.bot.service.BotService
import kotlinx.coroutines.runBlocking
import org.springframework.context.SmartLifecycle
import org.springframework.stereotype.Component

@Component
class BotLifeCycle(private val botService: BotService) : SmartLifecycle {

    override fun start() {
        runBlocking { botService.bot.login() }
        botService.registerHandlers()
    }

    override fun stop() {
        botService.bot.close()
    }

    override fun isRunning(): Boolean {
        return botService.bot.isOnline
    }
}
